From d3543e9140536a9751811b44f8496489be1e6bb7
From: Peter Stuge
Date: Mon, 13 Jun 2011 22:58:23 +0200
Subject: [PATCH] libusb.h: Add missing LIBUSB_CLASS_ codes currently defined by usb.org

LIBUSB_CLASS_PHYSICAL
LIBUSB_CLASS_SMART_CARD
LIBUSB_CLASS_CONTENT_SECURITY
LIBUSB_CLASS_VIDEO
LIBUSB_CLASS_PERSONAL_HEALTHCARE
LIBUSB_CLASS_DIAGNOSTIC_DEVICE

Per http://www.usb.org/developers/defined_class
---
 libusb/libusb.h |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/libusb/libusb.h b/libusb/libusb.h
index 7bb1b6a..45e20ec 100644
--- a/libusb/libusb.h
+++ b/libusb/libusb.h
@@ -155,6 +155,9 @@ enum libusb_class_code {
 	/** Human Interface Device class */
 	LIBUSB_CLASS_HID = 3,
 
+	/** Physical */
+	LIBUSB_CLASS_PHYSICAL = 5,
+
 	/** Printer dclass */
 	LIBUSB_CLASS_PRINTER = 7,
 
@@ -171,6 +174,21 @@ enum libusb_class_code {
 	/** Data class */
 	LIBUSB_CLASS_DATA = 10,
 
+	/** Smart Card */
+	LIBUSB_CLASS_SMART_CARD = 0x0b,
+
+	/** Content Security */
+	LIBUSB_CLASS_CONTENT_SECURITY = 0x0d,
+
+	/** Video */
+	LIBUSB_CLASS_VIDEO = 0x0e,
+
+	/** Personal Healthcare */
+	LIBUSB_CLASS_PERSONAL_HEALTHCARE = 0x0f,
+
+	/** Diagnostic Device */
+	LIBUSB_CLASS_DIAGNOSTIC_DEVICE = 0xdc,
+
 	/** Wireless class */
 	LIBUSB_CLASS_WIRELESS = 0xe0,
 
-- 
1.7.6.1

